src/backend/cpu/kernel/ireduce.hpp

```diff
@@ -49,13 +55,17 @@ struct MinMaxOp<af_max_t, T>
     MinMaxOp(T val, uint idx) :
         m_val(val), m_idx(idx)
     {
+        if (isNan(val)) {
+            m_val = Binary<T, af_max_t>::init();
+        }
     }
 
     void operator()(T val, uint idx)
     {
-        if (cabs(val) > cabs(m_val) ||
-            (cabs(val) == cabs(m_val) &&
-             idx <= m_idx)) {
+        if (!isNan(val) &&
```

Comment:
Switch order of isNan operation. See other comments for info.

Reply:
👍